src/EditorFeatures/Core.Wpf/NavigateTo/NavigateToItemDisplay.cs

```diff
@@ -114,16 +125,15 @@ public void NavigateTo()
             //
             // In the case of a stale item, don't require that the span be in bounds of the document
             // as it exists right now.
-            //
-            // TODO: Get the platform to use and pass us an operation context, or create one
-            // ourselves.
-            _threadingContext.JoinableTaskFactory.Run(() => navigationService.TryNavigateToSpanAsync(
+            using var context = _threadOperationExecutor.BeginExecute(
```

Comment:
Not *entirely* sure if I like the using this way since the scoping here is actually pretty critical.

Reply:
hrmm... this is pretty standard for how we use TWD in async contexts.  not sure i think there's a problem here.